courses/models_test.py

```diff
@@ -645,3 +705,197 @@ def test_program_certificate_start_end_dates_and_page_revision(user):
+                                    "course": course.id,
+                                    "node_type": ProgramRequirementNodeType.COURSE.value,
+                                },
+                                "id": node.id,
+                            }
+                            for course, node in zip(
+                                program_with_requirements.elective_courses,
+                                program_with_requirements.elective_courses_node.get_children(),
+                            )
+                        ],
+                        {
+                            "data": {
+                                **node_defaults,
+                                "operator": ProgramRequirement.Operator.MIN_NUMBER_OF.value,
+                                "operator_value": "1",
+                                "node_type": ProgramRequirementNodeType.OPERATOR.value,
+                            },
+                            "id": program_with_requirements.mut_exclusive_courses_node.id,
+                            "children": [
+                                {
+                                    "data": {
+                                        **node_defaults,
+                                        "course": course.id,
+                                        "node_type": ProgramRequirementNodeType.COURSE.value,
+                                    },
+                                    "id": node.id,
+                                }
+                                for course, node in zip(
+                                    program_with_requirements.mut_exclusive_courses,
+                                    program_with_requirements.mut_exclusive_courses_node.get_children(),
+                                )
+                            ],
+                        },
+                    ],
+                },
+            ],
+        }
+    ]
+
+
+@pytest.mark.parametrize(
+    "operator, expected",
+    [
+        (ProgramRequirement.Operator.ALL_OF.value, True),
+        (ProgramRequirement.Operator.MIN_NUMBER_OF.value, False),
+    ],
+)
+def test_program_requirements_is_all_of_operator(operator, expected):
+    """Test is_all_of_operator"""
+    assert ProgramRequirement(operator=operator).is_all_of_operator is expected
+
+
+@pytest.mark.parametrize(
+    "operator, expected",
+    [
+        (ProgramRequirement.Operator.ALL_OF.value, False),
+        (ProgramRequirement.Operator.MIN_NUMBER_OF.value, True),
+    ],
+)
+def test_program_requirements_is_min_number_of_operator(operator, expected):
+    """Test is_min_number_of_operator"""
+    assert ProgramRequirement(operator=operator).is_min_number_of_operator is expected
+
+
+def test_courses_in_program(program_with_requirements):
+    """Test CourseQuerySet.courses_in_program"""
+    CourseFactory.create_batch(4)
+
+    courses = Course.objects.courses_in_program(program_with_requirements.program)
```

Comment:
Is this test to ensure that an Admin can only add course requirements for courses that are associated with the program, or can the Admin add any course?

Reply:
No, this is to verify that the queryset function returns courses that show up in a program's requirements.